migrations/20200915113240_van-sync-task.js

```diff
@@ -4,28 +4,43 @@ exports.up = function(knex) {
     create function public.mark_van_sync_job_done(payload json, result json, context json)
       returns void as $$
     declare
+      v_job_request_id integer;
+      v_contact_count integer;
       v_remaining_syncs integer;
+      v_result_message text;
     begin
-      select count(*)
-      from graphile_worker.jobs jobs
-      where (jobs.payload->'__context'->>'job_request_id')::integer = (context->>'campaign_id')::integer
+      select
+        (context->>'job_request_id')::integer,
+        (context->>'contact_count')::integer
+      into v_job_request_id, v_contact_count;
+
+      with locked_rows as (
+        select 1
+        from graphile_worker.jobs jobs
+        where (jobs.payload->'__context'->>'job_request_id')::integer = v_job_request_id
+        for update
```

Comment:
This will block the jobs from executing for the duration of this function - I don't think we want that (I think it's better to have slightly inaccurate system progress), since this could be slow because of a delivery report backlog, etc.